dev/run-tests.py

```diff
@@ -571,8 +572,8 @@ def main():
     # backwards compatibility checks
     if build_tool == "sbt":
         # Note: compatibility tests only supported in sbt for now
-        detect_binary_inop_with_mima()
-        # Since we did not build assembly/assembly before running dev/mima, we need to
+        detect_binary_inop_with_mima(hadoop_version)
+        # Since we did not build assembly/package before running dev/mima, we need to
         # do it here because the tests still rely on it; see SPARK-13294 for details.
         build_spark_assembly_sbt(hadoop_version)
```

Comment:
The original motivation for this change was to ensure that MiMa would not accidentally rely on an assembly. Now that we're completely removing assemblies, does it make sense to go back to always unconditionally running `build_spark_assembly_*` before MiMA? Just wondering if this could simplify things a bit and also because this line comment is outdated / confusing now.